crux_macros/src/bridge.rs

```diff
@@ -0,0 +1,107 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::ItemStruct;
+
+pub fn bridge_impl(input: ItemStruct) -> TokenStream {
+    let ident = &input.ident;
+    quote! {
+        #[derive(Default)]
+        #input
+
+        static CORE: ::std::sync::LazyLock<::crux_core::bridge::Bridge<#ident>> =
+            ::std::sync::LazyLock::new(|| {
+                ::crux_core::bridge::Bridge::new(::crux_core::Core::new())
+            });
+
+        #[cfg(not(target_family = "wasm"))]
+        const _: () = assert!(
+            uniffi::check_compatible_version("0.29.2"),
+            "please use uniffi v0.29.2"
+        );
+
+        #[cfg(not(target_family = "wasm"))]
+        ::uniffi::setup_scaffolding!();
+
+        #[cfg_attr(not(target_family = "wasm"), ::uniffi::export)]
+        #[cfg_attr(target_family = "wasm", ::wasm_bindgen::prelude::wasm_bindgen)]
+        pub fn process_event(data: &[u8]) -> Vec<u8> {
+            match CORE.process_event(data) {
+                Ok(effects) => effects,
+                Err(e) => panic!("{e}"),
+            }
+        }
```

Comment:
More of a uniffi noob question: Are the exports indpendent of module path? Does it matter where the macro is used?

And am I assuming correctly that this can only be used once per crate?